onnx/defs/tensor/defs.cc

```diff
@@ -240,7 +240,36 @@ will be (2, 1, 3).
     .TypeConstraint(
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
-        "Constrain input and output types to float tensors.");
+        "Constrain input and output types to float tensors.")
+    .ShapeInferenceFunction([](InferenceContext& ctx) {
+    std::vector<int64_t> perm;
+    {
+      auto perm_attr = ctx.getAttribute("perm");
+      if (perm_attr) {
+        for (int i = 0; i < perm_attr->ints_size(); i++) {
+          perm.push_back(perm_attr->ints(i));
+        }
+      } else {
+        int ndims = ctx.getAllInputTypes()[0].shape().dim_size();
+        for (int i = 0; i < ndims; i++) {
+          perm.push_back(ndims - i - 1);
+        }
+      }
+    }
+
+    TypeProto_Tensor tt;
```

Comment:
shall we have some helper function for creating TensorProto_Tensor too? otherwise this mechanics of protobuf are too cumbersome to operate.

Reply:
sure, but same as above - this is a small thing, we can iterate on it.